src/containers/List/ListFilters/ListFilters.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { useEffect, useState } from 'react'
+import { Dropdown } from 'semantic-ui-react'
+import { useRouter } from '../../../utils/hooks/useRouter'
+import { FilterDefinitions } from '../../../utils/types'
+import BooleanFilter from './BooleanFilter'
+import { EnumFilter, SearchableListFilter, StaticListFilter } from './OptionFilters'
+import { GetMethodsForOptionFilter } from './types'
+
+const getArrayFromString = (string: string = '') =>
+  string.split(',').filter((option) => !!option.trim())
+
+const getDisplayableFilters = (filterDefinitions: FilterDefinitions) => {
+  const result: FilterDefinitions = {}
+  Object.entries(filterDefinitions).forEach(([filterName, filterValue]) => {
+    // Missing title determines if filters should be shown in UI or not
+    if (!filterValue.title) return
+    result[filterName] = filterValue
+  })
+
+  return result
+}
+
+const ListFilters: React.FC<{ filterDefinitions: FilterDefinitions; filterListParameters: any }> =
+  ({ filterDefinitions, filterListParameters }) => {
+    const { query, updateQuery } = useRouter()
+
+    const displayableFilters = getDisplayableFilters(filterDefinitions)
+
+    const filterNames = Object.keys(displayableFilters)
+    const [activeFilters, setActiveFilters] = useState<string[]>([])
+
+    // Add fitlers from URL to activeFilters, filters may not have criteria/options yet thus we have to use activeFilter state variable
+    useEffect(() => {
+      const filersInQuery = filterNames.filter((filterName) => !!query[filterName])
+      const newFilters = filersInQuery.filter((filterName) => !activeFilters.includes(filterName))
+      setActiveFilters([...activeFilters, ...newFilters])
+    }, [query])
+
+    // Filter criteria/options states is provided by query URL, methods below are get and set query fiter criteria
+    const getMethodsForOptionFilter: GetMethodsForOptionFilter = (filterName) => ({
+      getActiveOptions: () => getArrayFromString(query[filterName]),
+      setActiveOption: (option: string) =>
+        updateQuery({ [filterName]: [...getArrayFromString(query[filterName]), option].join(',') }),
+      setInactiveOption: (option: string) =>
+        updateQuery({
+          [filterName]: getArrayFromString(query[filterName])
+            .filter((_option) => _option != option)
+            .join(','),
+        }),
+    })
+
+    const addFilter = (filterName: string) => setActiveFilters([...activeFilters, filterName])
+    const getOnRemove = (filterName: string) => () => {
```

Comment:
not sure what this function meant - is this used when removing all options for a given filterName?

Reply:
Yeah, it's for removing the whole filter option, and the `get` part of it, so we don't have to do the following in each filter instantiation below
```
  setActiveFilters(activeFilters.filter((_filterName) => _filterName !== filterName))
      updateQuery({
        [filterName]: undefined,
      })
```